File: frameworks/library_media/src/main/java/com/wangzhumo/app/mdeia/gles/OffscreenSurface.java
package com.wangzhumo.app.mdeia.gles;

/**
 * If you have any questions, you can contact by email {wangzhumoo@gmail.com}
 *
 * @author 王诛魔 2020-01-07  11:03
 *
 * Off-screen EGL surface (pbuffer).
 */
public class OffscreenSurface extends EGLSurfaceBase {


    /**
     * Creates an off-screen surface with the specified width and height.
     */
    public OffscreenSurface(EGLCore eglCore,int width, int height) {
        super(eglCore);
        createOffscreenSurface(width, height);
    }

    /**
     * Releases any resources associated with the surface.
     */
    public void release() {
        releaseEglSurface();
    }
}

File: module_media/src/main/java/com/wangzhumo/app/module/media/opengl/gles/OffscreenSurface.kt
package com.wangzhumo.app.module.media.opengl.gles

/**
 * If you have any questions, you can contact by email {wangzhumoo@gmail.com}
 *
 * @author 王诛魔 2019-12-31  16:13
 *
 * Off-screen EGL surface (pbuffer).
 */
class OffscreenSurface constructor(eglHelper: EGLHelper,width :Int,height :Int) : EglSurfaceBase(eglHelper) {

    init {
        createOffscreenSurface(width, height)
    }

    /**
     * Releases any resources associated with the surface.
     */
    fun release() {
        releaseEglSurface()
    }

}